runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DockerJobBundleFactory.java

```diff
@@ -135,7 +135,6 @@ public static DockerJobBundleFactory create(JobInfo jobInfo)
     this.provisioningServer = provisioningServer;
     this.environmentCache =
         CacheBuilder.newBuilder()
-            .weakValues()
```

Comment:
We explicitly call invalidateAll in https://github.com/apache/beam/pull/5692/files#diff-b74ff692340bcae0032d119a7192624cL181 so no need to have weekValues

Reply:
We still want weak values. Otherwise, all containers started by this job will stick around until the job is finished.

Reply:
Yes, containers should stick around for the jobDuration so that we can reuse them. 
Currently it sticks around for as long as BatchFlinkExecutableStageContext sticks around and BatchFlinkExecutableStageContext is active till there is an ongoing Function request for the associated job.
Is there a reason to evict container sooner?